buffer/src/main/java/io/netty/buffer/AbstractUnsafeSwappedByteBuf.java

```diff
@@ -76,21 +76,21 @@ public final int getUnsignedShort(int index) {
 
     @Override
     public final short getShort(int index) {
-        wrapped.checkIndex0(index, 2);
+        wrapped.checkIndex(index, 2);
         short v = _getShort(wrapped, index);
         return nativeByteOrder ? v : Short.reverseBytes(v);
     }
 
     @Override
     public final ByteBuf setShort(int index, int value) {
-        wrapped.checkIndex0(index, 2);
+        wrapped.checkIndex(index, 2);
```

Comment:
@fenik17 why this change ?

Reply:
this looks unrelated to the PR.

Reply:
Without this fixes some unit tests is failed. I described this in `Modifications` part (see the PR description). Should I extract this in a separate PR?

Reply:
ah got it... sorry missed it